scaffolder doesn't use plural from DbSet

new shorter instructions

1.  Create a RP web app named `RazorPagesMovie`
1. Follow [these](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model) instructions **EXCEPT** - name the `DbSet` plural
`public DbSet<Movie> Movies { get; set; }`

This tutorial had 30K confirming it works until an update was made to change to the plural `public DbSet<Movie> Movies { get; set; }`

-- what follows is essentially the same instructions

In VS, create a new Razor Pages (RP) app named ContosoUniversity
Create a Models folder. In the Models folder, create a class file named Student.cs and replace the code with the following:
```
using System;
using System.Collections.Generic;

namespace ContosoUniversity.Models
{
    public class Student
    {
        public int ID { get; set; }
        public string LastName { get; set; }
        public string FirstMidName { get; set; }
        public DateTime EnrollmentDate { get; set; }
    }
}
```

In the Data folder create a new class file named SchoolContext.cs, and replace the template code with the following code:

```
using ContosoUniversity.Models;
using Microsoft.EntityFrameworkCore;

namespace ContosoUniversity.Data
{
    public class SchoolContext : DbContext
    {
        public SchoolContext(DbContextOptions<SchoolContext> options) : base(options)
        {
        }

        public DbSet<Student> Students { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {

            modelBuilder.Entity<Student>().ToTable("Student");
        }
    }
}
```

Add SchoolContext to the DI container:

```
// ADD EF using
// using Microsoft.EntityFrameworkCore;

public void ConfigureServices(IServiceCollection services)
{
    services.AddDbContext<Data.SchoolContext>(options =>
        options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));

    services.AddMvc();
}
```

Open the appsettings.json file and add a connection string as shown in the following code:

```
{
  "ConnectionStrings": {
    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=ContosoUniversity1;Trusted_Connection=True;MultipleActiveResultSets=true"
  },
  "Logging": {
    "IncludeScopes": false,
    "LogLevel": {
      "Default": "Warning"
    }
  }
}
```
In the Package Manager Console (PMC), enter the following commands:
```
Install-Package Microsoft.VisualStudio.Web.CodeGeneration.Design -Version 2.0.0
Add-Migration Initial
Update-Database
```
*Scaffold the model

dotnet aspnet-codegenerator razorpage -m Student -dc SchoolContext -udl -outDir Pages\CU --referenceScriptLibraries

Build the project. You get errors
` error CS1061: 'SchoolContext' does not contain a definition for 'Student' and no extension method 'Student'`

Here is the generated code: `            _context.Student.Add(Student);
`
It should be plural 
`_context.Students.Add(Student);`

--- a/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/CreatePageModel.cshtml
+++ b/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/CreatePageModel.cshtml
@@ -59,7 +59,7 @@ namespace @Model.NamespaceName
                 return Page();
             }
 
-            _context.@(Model.ModelTypeName).Add(@Model.ViewDataTypeShortName);
+            _context.@(entitySetName).Add(@Model.ViewDataTypeShortName);
             await _context.SaveChangesAsync();
 
             return RedirectToPage("./Index");

--- a/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/DeletePageModel.cshtml
+++ b/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/DeletePageModel.cshtml
@@ -9,6 +9,7 @@ using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.EntityFrameworkCore;
 @{
+    var entitySetName = Model.ModelMetadata.EntitySetName;
     var primaryKeyName = Model.ModelMetadata.PrimaryKeys[0].PropertyName;
     var primaryKeyShortTypeName = Model.ModelMetadata.PrimaryKeys[0].ShortTypeName;
     var primaryKeyType = Model.ModelMetadata.PrimaryKeys[0].TypeName;
@@ -53,7 +54,7 @@ namespace @Model.NamespaceName
                 return NotFound();
             }
 
-            @Model.ViewDataTypeShortName = await _context.@(Model.ModelTypeName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@primaryKeyName == id);
+            @Model.ViewDataTypeShortName = await _context.@(entitySetName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@primaryKeyName == id);
 
             if (@Model.ViewDataTypeShortName == null)
             {
@@ -69,11 +70,11 @@ namespace @Model.NamespaceName
                 return NotFound();
             }
 
-            @Model.ViewDataTypeShortName = await _context.@(Model.ModelTypeName).FindAsync(id);
+            @Model.ViewDataTypeShortName = await _context.@(entitySetName).FindAsync(id);
 
             if (@Model.ViewDataTypeShortName != null)
             {
-                _context.@(Model.ModelTypeName).Remove(@Model.ViewDataTypeShortName);
+                _context.@(entitySetName).Remove(@Model.ViewDataTypeShortName);
                 await _context.SaveChangesAsync();
             }
 

--- a/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/DetailsPageModel.cshtml
+++ b/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/DetailsPageModel.cshtml
@@ -9,6 +9,7 @@ using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.EntityFrameworkCore;
 @{
+    var entitySetName = Model.ModelMetadata.EntitySetName;
     var primaryKeyName = Model.ModelMetadata.PrimaryKeys[0].PropertyName;
     var primaryKeyShortTypeName = Model.ModelMetadata.PrimaryKeys[0].ShortTypeName;
     var primaryKeyType = Model.ModelMetadata.PrimaryKeys[0].TypeName;
@@ -52,7 +53,7 @@ namespace @Model.NamespaceName
                 return NotFound();
             }
 
-            @Model.ViewDataTypeShortName = await _context.@(Model.ModelTypeName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@(primaryKeyName) == id);
+            @Model.ViewDataTypeShortName = await _context.@(entitySetName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@(primaryKeyName) == id);
 
             if (@Model.ViewDataTypeShortName == null)
             {

--- a/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/EditPageModel.cshtml
+++ b/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/EditPageModel.cshtml
@@ -55,7 +55,7 @@ namespace @Model.NamespaceName
                 return NotFound();
             }
 
-            @Model.ViewDataTypeShortName = await _context.@(Model.ModelTypeName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@(primaryKeyName) == id);
+            @Model.ViewDataTypeShortName = await _context.@(entitySetName)@(inlineIncludes).SingleOrDefaultAsync(m => m.@(primaryKeyName) == id);
 
             if (@Model.ViewDataTypeShortName == null)
             {

--- a/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/ListPageModel.cshtml
+++ b/src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Templates/RazorPageGenerator/ListPageModel.cshtml
@@ -9,6 +9,7 @@ using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.EntityFrameworkCore;
 @{
+    var entitySetName = Model.ModelMetadata.EntitySetName;
     var primaryKeyName = Model.ModelMetadata.PrimaryKeys[0].PropertyName;
     var primaryKeyShortTypeName = Model.ModelMetadata.PrimaryKeys[0].ShortTypeName;
     var primaryKeyType = Model.ModelMetadata.PrimaryKeys[0].TypeName;
@@ -46,7 +47,7 @@ namespace @Model.NamespaceName
 
         public async Task OnGetAsync()
         {
-            @Model.ViewDataTypeShortName = await _context.@(Model.ModelTypeName)@(inlineIncludes).ToListAsync();
+            @Model.ViewDataTypeShortName = await _context.@(entitySetName)@(inlineIncludes).ToListAsync();
         }
     }
 }

